### PR TITLE
New error code for missing well-known ValueTuple type

### PR DESCRIPTION
**Pull request by jcouv**
This is a re-work of PR https://github.com/dotnet/roslyn/pull/13542.
Aleksey pointed out that the new error should only appear when looking up the ValueTuple well-known types, not for other missing references to ValueTuple. This simplifies the logic, both for C# and VB.

The purpose is to report a distinct error code so that Cyrus's code fixer can trigger easily and recommend to add the ValueTuple nuget package.

@dotnet/roslyn-compiler for review.
FYI for @CyrusNajmabadi 
Relates to #13375

### CODE REVIEW

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 1]
```diff
@@ -588,6 +588,12 @@ public static bool IsWellKnownType(this WellKnownType typeId)
             return typeId >= WellKnownType.First && typeId < WellKnownType.NextAvailable;
         }
 
+        public static bool IsValueTupleType(this WellKnownType typeId)
+        {
+            Debug.Assert(typeId != WellKnownType.ExtSentinel);
+            return typeId >= WellKnownType.System_ValueTuple_T1 && typeId < WellKnownType.System_ValueTuple_TRest;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: >< [](start = 74, length = 1)

Should this be <=?

File: src/Compilers/VisualBasic/Portable/VBResources.resx [hunk 2]
```diff
@@ -1995,6 +1995,9 @@
   <data name="ERR_TypeRefResolutionError3" xml:space="preserve">
     <value>Import of type '{0}' from assembly or module '{1}' failed.</value>
   </data>
+  <data name="ERR_ValueTupleTypeRefResolutionError" xml:space="preserve">
+    <value>Import of type '{0}' failed.</value>
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/VBResources.resx]: >Import of type '{0}' failed. [](start = 11, length = 28)

The message feels somewhat misleading, there was no actual import failure. Consider using something like "Unable to locate type {0}."

File: src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs [hunk 3]
```diff
@@ -128,6 +136,14 @@ internal NamedTypeSymbol GetWellKnownType(WellKnownType type)
             return _lazyWellKnownTypes[index];
         }
 
+        private CSDiagnosticInfo CreateValueTupleTypeNotFoundError(MetadataTypeName name)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs]: `static`. Or perhaps inline the method.

Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs]: Inlined

File: src/Compilers/Core/Portable/WellKnownTypes.cs (hunk 1 above)
Comment (jcouv) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: I apparently forgot to run that test :-P

**Comment by cston** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/VBResources.resx (hunk 2 above)
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/VBResources.resx]: New message is like C#:
"BC37267: Predefined type 'ValueTuple(Of ,)' is not defined or imported."

**Comment by AlekseyTs** [comment]
LGTM

**Comment by jcouv** [comment]
Thanks for the reviews. I'll go ahead and merge.

**Pull request merged**